scipy/optimize/_hessian_update_strategy.py

```diff
@@ -190,12 +190,18 @@ def update(self, delta_x, delta_grad):
             if self.init_scale == "auto":
                 scale = self._auto_scale(delta_x, delta_grad)
             else:
-                scale = float(self.init_scale)
+                scale = self.init_scale
```

Comment:
if for some reason complex numbers with zero imaginary component comes they will upcast everything

Reply:
Fair point, and numpy is quite inconsistent here I figured:
```
np.array(complex(42), dtype=np.float64)  # raises TypeError
np.array(complex_array, dtype=np.float64)  # casts silently
```
So I've added an explicit check to have the same behavior as before in case of complex objects: a TypeError is raised

Reply:
I would check if it is an ndarray, its `.size == 1` and if it is one push it to `float()` otherwise continue with the matrix case.

Reply:
> I would check if it is an ndarray, its `.size == 1` and if it is one push it to `float()` otherwise continue with the matrix case.

Why would you do that? To me this seems just like extra code that won't change any of the logic. Or are you worried about implications of removing the `float` cast?

Reply:
But sure, can be done, added it

Reply:
No we don't want to stop the complex data leading to errors. That's backwards incompatible, what you are adding shouldn't touch the old behavior. Test for an instance of numpy array then if it is a scalar (from testing size ==1) it passes through float otherwise it is treated as the full hessian and continues.

Reply:
Yes, I've added that, basically simply testing if it is size 1 -> old behavior else new

Reply:
I've clarified below